example/JJMasterData.Web.Example/Program.cs

```diff
@@ -13,9 +13,9 @@ public static void Main(string[] args)
 
         var root = Path.Join(builder.Environment.ContentRootPath, "..", "..");
         var settingsPath = Path.Combine(root, "appsettings.json");
-        
-        builder.Configuration.AddJsonFile(settingsPath, false, true);
 
+        builder.Configuration.AddJsonFile(settingsPath, false, true);
+        
         builder.Services.AddRazorPages();
         builder.Services.AddControllersWithViews();
 
```

Comment:
Unfortanally this is impossible 😥. Because the method now have a reference to `builder.Configuration` in these lines:

```cs
        builder.Services.ConfigureOptionsWriter<JJMasterDataOptions>(
            builder.Configuration.GetSection("JJMasterData"), settingsPath);
        

        builder.Services.ConfigureOptionsWriter<ConnectionStrings>(
            builder.Configuration.GetSection("ConnectionStrings"), settingsPath);

        builder.Services.ConfigureOptionsWriter<ConnectionProviders>(
            builder.Configuration.GetSection("ConnectionProviders"), settingsPath);
```
We have two options:

1.
```cs
builder.Services.AddJJMasterDataWeb(builder.Configuration)
```
2. 
```cs
builder.ConfigureJJMasterDataOptionsWriters()
builder.Services.AddJJMasterDataWeb()
```

Reply:
In my opinion the configuration should stay out of this method and I didn't quite understand why injecting, if we can directly call the OptionsWriter class...

Reply:
> In my opinion the configuration should stay out of this method and I didn't quite understand why injecting, if we can directly call the OptionsWriter class...

There is two reasons to inject

1. Someone would need to change OptionsWriter to write to another `IOptions` source, like static variables or XML files (first-class supported when adding configuration sources).
2. The class have a `IOptionsMonitor<T>` dependency and we don't know how to instantiate this, because of that, we need to use DI.

But I agree we should split from MasterData services, the MasterData configuration layer.